frontend/src/scenes/retention/retentionTableLogic.ts

```diff
@@ -73,28 +73,42 @@ export const retentionTableLogic = kea<retentionTableLogicType>([
                 const { period } = retentionFilter || {}
                 const { breakdowns } = breakdownFilter || {}
 
-                return range(maxIntervalsCount).map((rowIndex: number) => [
-                    // First column is the cohort label
-                    breakdowns?.length
-                        ? results[rowIndex].label
-                        : period === 'Hour'
-                        ? dayjs(results[rowIndex].date).format('MMM D, h A')
-                        : dayjs(results[rowIndex].date).format('MMM D'),
-                    // Second column is the first value (which is essentially the total)
-                    ...(hideSizeColumn ? [] : [results[rowIndex].values[0].count]),
-                    // All other columns are rendered as percentage
-                    ...results[rowIndex].values.map((row) => {
-                        const percentage =
-                            results[rowIndex].values[0]['count'] > 0
-                                ? (row['count'] / results[rowIndex].values[0]['count']) * 100
-                                : 0
+                return range(maxIntervalsCount).map((index: number) => {
+                    const currentResult = results[index]
+                    let firstColumn // Prepare for some date gymnastics
+
+                    if (breakdowns?.length) {
+                        firstColumn = currentResult.label
+                    } else {
+                        switch (period) {
+                            case 'Hour':
+                                firstColumn = dayjs(currentResult.date).format('MMM D, h A')
+                                break
+                            case 'Month':
+                                firstColumn = dayjs(currentResult.date).format('MMM YYYY')
+                                break
+                            case 'Week':
+                                firstColumn = `Week of ${dayjs(currentResult.date).format('MMM D')}`
```

Comment:
The small problem with this is if it's "Week of [day that's Sunday]", Americans will believe that's when the week starts, but Europeans will assume that's when the week ends. So maybe "week starting..." makes it even clearer?

Reply:
Makes sense. But we do take into account the week start day setting here or don't we? I read about it but haven't tested.

Reply:
I'm.... not too sure 😅. Probably not. But if not, we should at _some_ point...

Reply:
It works for me... can change from Monday to Sunday 🎉